### ISSUE TITLE
Unused Argument exception lead to problems

### ISSUE THREAD

**Issue by Zeromax** [contao/core#7659]
Since #5570 the `UnusedArgumentsException` is implemented and causes the page to render twice.

There we have a `$GLOBALS` Problem.
For Example:
My page renders regular page and added some Options to `$GLOBALS['TL_HEAD']`.
if the Exception is called the `$GLOBALS['TL_HEAD']` should be set to empty as well as some other `$GLOBALS` Array for Javascript and so on.

This can lead to XSS Injection Problems. Which I have discovered with `changelanguage`.

Discovered in LTS -> https://github.com/contao/core/blob/3.2.19/index.php#L255

**Comment by leofeyer** [comment]
As discussed on Mumble on February 26th, we should try to buffer the state of the global variables as used in `Controller::replaceDynamicScriptTags()` before the try-catch-block in the `FrontendIndex` class:

```php
$globalHead = $GLOBALS['TL_HEAD']; // <-- Buffer

try
{
	// Generate the page
	switch ($objPage->type)
	{
		case 'root':
		case 'error_404':
			$objHandler->generate($pageId);
			break;

		case 'error_403':
			$objHandler->generate($pageId, $objRootPage);
			break;

		default:
			$objHandler->generate($objPage, true);
			break;
	}
}
catch (\UnusedArgumentsException $e)
{
	$GLOBALS['TL_HEAD'] = $globalHead; // <-- reset

	// Render the error page (see #5570)
	$objHandler = new $GLOBALS['TL_PTY']['error_404']();
	$objHandler->generate($pageId, null, null, true);
}
```

**Comment by leofeyer** [comment]
Can you please point me again to how to reproduce this? I don't remember.

**Comment by Zeromax** [comment]
1. Add something to the `$GLOBALS['TL_HEAD']`
2. Call the page with a unused get param, can be a auto_item or what ever
3. In the `$GLOBALS['TL_HEAD']` all stuff from the first page Render is available and from the second as well.

**Comment by leofeyer** [comment]
Where do I add something? In a `config.php` file? Or in the "additional HEAD tags" field of the page layout?

**Comment by Zeromax** [comment]
In a config File.

Not Sure if you get the same result with the page layout.

**Comment by discordier** [comment]
I guess you won't get the same result in the page layout.

**Comment by leofeyer** [comment]
Fixed in 6bde4b76a7392c765c6fa72bf627c96949d96578.
